### PR TITLE
[amberscript] Parse PIPELINE BIND calls for color and depth buffers

### PR DESCRIPTION
**Pull request by dj2**
This CL adds parsing for the AmberScript color and depth attachments. It
also updates the pipelines to create a default color or depth attachment
if one is not provided.

### CODE REVIEW

**Comment by jaebaek** [comment]
I have two more comments for amber script, not this CL:

1. It would be better to remove "sampled" from "Buffer Types". Vulkan engine does not support it now and we do not have specific needs / design for it now.

2. We should remove the following as well for the same reason
```
# Bind the sampler at the given descriptor set and binding.
  BIND SAMPLER <sampler_name> DESCRIPTOR_SET <id> BINDING <id>
```

File: docs/amber_script.md [hunk 1]
```diff
@@ -180,14 +180,16 @@ A `pipeline` can have buffers bound. This includes buffers to contain image
 attachment content, depth/stencil content, uniform buffers, etc.
 
 ```
-  # Attach |buffer_name| as an output colour attachment at location |idx|.
-  # The provided buffer must be a `FORMAT` buffer. If no colour attachments are
-  # provided a single attachment with format `R8G8B8A8_UINT` will be created.
+  # Attach |buffer_name| as an output color attachment at location |idx|.
+  # The provided buffer must be a `FORMAT` buffer. If no color attachments are
+  # provided a single attachment with format `R8G8B8A8_UINT` will be created
```
[the remark this answers is not in the document]
Comment (dj2) [reply docs/amber_script.md]: Yup, looks like I got the wrong one. Will update to be B8G8R8A8_UNORM.

**Comment by dj2** [comment]
Even though we don't have it now, we'll want sampled in the future so we'll want amberscript to parse it even if it isn't supported by the engine at this point. We'll just have the engine return an unsupported error.

File: docs/amber_script.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (dj2) [reply docs/amber_script.md]: done.

**Pull request merged**